File: specification/specs/search/search/hits/HitMetadata.ts
class HitMetadata<TDocument> {
	_id: string;
	_index: string;
	_primary_term: long;
	_routing: string;
	_seq_no: long;
	@prop_serializer("SourceFormatter`1")
	_source: TDocument;
	_type: string;
	_version: long;
}

File: output/csharp/Search/Search/Hits/HitMetadata.cs
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Search {

	public class HitMetadata<TDocument>  {
		
		[DataMember(Name="_id")]
		public string Id { get; set; }


		[DataMember(Name="_index")]
		public string Index { get; set; }


		[DataMember(Name="_primary_term")]
		public long PrimaryTerm { get; set; }


		[DataMember(Name="_routing")]
		public string Routing { get; set; }


		[DataMember(Name="_seq_no")]
		public long SeqNo { get; set; }


		[DataMember(Name="_source")]
		public TDocument Source { get; set; }


		[DataMember(Name="_type")]
		public string Type { get; set; }


		[DataMember(Name="_version")]
		public long Version { get; set; }

	}
}
